one_fm/api/dashboard_utils.py

```diff
@@ -47,4 +50,109 @@ def get_open_count(doctype, name, links):
 	if hasattr(module, 'get_timeline_data'):
 		out['timeline_data'] = module.get_timeline_data(doctype, name)
 	# print(out)
-	return out
\ No newline at end of file
+	return out
+
+
+@frappe.whitelist()
+def get_employee_shift(employee_id, date_type='month'):
+	"""
+		Fetch employee information relating to
+		leave_balance, shift, penalties and attendance
+	"""
+
+	# prepare dates
+	today = datetime.today()
+	_start = today.replace(day=1).date()
+	_end = today.replace(day = calendar.monthrange(today.year, today.month)[1]).date()
+	if(date_type=='year'):
+		_start = date(today.year, 1, 31)
+		_end = date(today.year, 12, 31)
+
+	data = {
+		"leave_balance": "",
+		"penalties": "",
+		"days_worked": "",
+		"shift_time_from": "",
+		"shift_time_to": "",
+		"shift_location": "",
+		"shift_latitude_and_longitude": ""
+	}
+	try:
+		employee = frappe.get_doc('Employee', employee_id)
+		shift_assignment =  frappe.db.get_list('Shift Assignment',
+     		filters=[
+		         ['employee', '=', employee.name],
+		     ],
+		     fields=['name', 'employee', 'site', 'shift_type'],
+		     order_by='modified desc',
+		     limit=1,
+		 )
+		if(len(shift_assignment)):
+			shift_location = frappe.get_doc('Location', shift_assignment[0].site)
+			shift_type = frappe.get_doc('Shift Type', shift_assignment[0].shift_type)
+			days_worked = frappe.db.get_list('Attendance',
+		    filters=[
+	             ['employee', '=', employee.name],
+	             ['status', '=', 'Present'],
+	             ['attendance_date', 'BETWEEN', [_start, _end]],
+	        ],
+		    fields=['COUNT(*) as count', 'name', 'employee', 'status', 'attendance_date'],
+		    order_by='modified desc',
+		 	)[0].count
+			penalties = frappe.db.sql(f"""
+				SELECT COUNT(*) as count, pi.name, pie.employee_id

```

Comment:
This query might not run without a GROUP BY clause. Just confirm it's working

Reply:
'COUNT(*) as count' automatically groups the result.

![image](https://user-images.githubusercontent.com/10146518/147937767-ba9be566-4d80-4d08-bdb4-21f71c766478.png)

Reply:
Great. Thanks